test/e2e/testsuites/metrics.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+Copyright 2022 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	"github.com/google/uuid"
+	metricspkg "github.com/googlecloudplatform/gcs-fuse-csi-driver/pkg/metrics"
+	"github.com/googlecloudplatform/gcs-fuse-csi-driver/test/e2e/specs"
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	dto "github.com/prometheus/client_model/go"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2evolume "k8s.io/kubernetes/test/e2e/framework/volume"
+	storageframework "k8s.io/kubernetes/test/e2e/storage/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var expectedMetricNames = []string{
+	"fs_ops_count",
+	"fs_ops_error_count",
+	"fs_ops_latency",
+	"gcs_download_bytes_count",
+	"gcs_read_count",
+	// "gcs_read_bytes_count", // TODO: cannot find these metrics
```

Comment:
Hi @sethiay, in my e2e tests, I don't see these metrics, so what operation will trigger these metrics?

Reply:
reading a file should trigger these metrics.

Reply:
@sethiay yes, in this e2e test I am doing `ls`, `touch`, `cat`, and `rm` commands against the mounted volume, but I still don't see these metrics. Line 140 - 146 are the test steps. Could you help check if I am missing anything?

Reply:
@songjiaxun I checked for one of the commented out metrics i.e. gcs_reader_count and I am able to see the reader_count exported via stackdriver on console but not when I do `curl http://localhost:8080/metrics` as mentioned in description of https://github.com/GoogleCloudPlatform/gcsfuse/pull/1859, then I didn't get it.

CC' @kislaykishore

Reply:
I missed one line change to enable these metrics in the gcsfuse PR, modified the PR: https://github.com/GoogleCloudPlatform/gcsfuse/pull/1859/files#diff-57454978dcd397e797f4c48e4b1f80b4b249655d5fc975ad55c54d560ba01de1